src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/QueryBuilder.scala

```diff
@@ -193,8 +193,21 @@ private[querygeneration] class QueryBuilder(plan: LogicalPlan) {
           }
         }
 
-      case Union(children) =>
-        Some(UnionQuery(children, alias.next))
+      // From Spark 3.1, Union has 3 parameters
```

Comment:
Could you write an explanation (as a comment) as to why this isn't supported in Snowflake? What do `byName` and `allowMissingCol` do?